src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java

```diff
@@ -0,0 +1,120 @@
+            preparedStatement.close();
+            connection.close();
+            return Optional.ofNullable(manufacturer);
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get manufacturer by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String query = "SELECT * FROM manufacturer WHERE is_deleted = false;";
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultset = preparedStatement.executeQuery();
+            while (resultset.next()) {
+                manufacturers.add(createManufacturer(resultset));
+            }
+            preparedStatement.close();
+            connection.close();
+        } catch (SQLException ex) {
+            throw new RuntimeException("Can't getAll manufacturers ", ex);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String query = "UPDATE manufacturer SET manufacturer_name=?, manufacturer_country=?"
+                + " WHERE manufacturer_id =?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.setLong(3, manufacturer.getId());
+            preparedStatement.executeUpdate();
+            preparedStatement.close();
+            connection.close();
+        } catch (SQLException ex) {
+            throw new RuntimeException("Can't update manufacturer " + manufacturer, ex);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String sqlQuery = "UPDATE manufacturer"
+                + " SET is_deleted = TRUE"
+                + " WHERE manufacturer_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sqlQuery)) {
+            preparedStatement.setLong(1, id);
+            int updatesMade = preparedStatement.executeUpdate();
+            preparedStatement.close();
+            connection.close();
+            return updatesMade > 0;
+        } catch (SQLException ex) {
+            throw new RuntimeException("Can't delete manufacturer by id " + id, ex);
+        }
+    }
+
+    private Manufacturer createManufacturer(ResultSet resultset) throws SQLException {
+        Long id = resultset.getObject("manufacturer_id", Long.class);
+        String name = resultset.getObject("manufacturer_name", String.class);
+        String country = resultset.getObject("manufacturer_country", String.class);
```

Comment:
here, you can use getString();

Reply:
Here I just follow checklist recomendation:
[https://prnt.sc/wo1ydy](url)